core/src/main/java/com/klaviyo/coresdk/KlaviyoConfig.kt

```diff
@@ -0,0 +1,58 @@
+package com.klaviyo.coresdk
+
+import java.lang.Exception
+
+class KlaviyoMissingAPIKeyException: Exception("You must declare an API key for the Klaviyo SDK")
+
+class KlaviyoConfig private constructor(
+        apiKey: String,
+        networkTimeout: Int,
+        networkFlushInterval: Int) {
+
+    val apiKey: String
+    val networkTimeout: Int
+    val networkFlushInterval: Int
+
+    init {
+        if (apiKey.isNullOrEmpty()) {
+            throw KlaviyoMissingAPIKeyException()
+        }
+        if (networkTimeout < 0) {
+            // TODO: When Timber is installed, log warning here
```

Comment:
should we default here?

Reply:
Defaults are defined by the builder, so this is just a catch for if the user overrides the default with their own invalid value

Reply:
right but the building isnt doing any validation besides the api key, so the network values will still be negative

Reply:
Oh of course, totally overlooked what you were getting at. Fixing now. And I'll add a test case to verify